felix/proto/felixbackend.proto

```diff
@@ -568,6 +568,9 @@ message RouteUpdate {
   bool nat_outgoing = 8;
   bool local_workload = 9;
   TunnelType tunnel_type = 10;
+  map<string, string> labels = 11;
```

Comment:
I'd like to discuss this one with @coutinhop and @fasaxc - it seems incongruous to be including things like node labels, node AS number, and wireguard public key on a RouteUpdate message since those aren't really attributes of a route itself.

I understand that this object already has a few somewhat questionable fields (e.g., ip_pool_type) and that we want to avoid unnecessary calculations in the data plane. But conceptually, tying these to a route update feels wrong and will result in sending substantially more information (repeating labels across routes rather than just once per node as needed).

We already send WireguardPublicKey to the data plane it seems:

- https://github.com/projectcalico/calico/blob/3f3d6b353d70695dc2d5ff6a64d40773a2b9847a/felix/proto/felixbackend.proto#L457

The node labels and AS number I would expect to be tied to a message representing a node object. For example, the HostMetadataUpdate object.
 https://github.com/projectcalico/calico/blob/3f3d6b353d70695dc2d5ff6a64d40773a2b9847a/felix/proto/felixbackend.proto#L465

Reply:
I agree with @caseydavenport, it makes more sense to add AS number and labels to HostMetadataUpdate and subscribe to it + WireguardStatusUpdate wherever you need this information instead of adding everything to RouteUpdate...

Reply:
https://github.com/projectcalico/calico/pull/7042
Here's a new PR with the new logic.